sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -543,4 +544,18 @@ public TableServiceClientBuilder clientOptions(ClientOptions clientOptions) {
 
         return this;
     }
+
+    /**
+     * Sets a flag indicating if tenant discovery should be enabled when authenticating with the Storage Table Service.
+     * This is set to {@code false} by default.
+     *
+     * @param enabled Value indicating whether tenant discovery is enabled or not.
+     *
+     * @return The updated {@link TableServiceClientBuilder}.
+     */
+    public TableServiceClientBuilder enableTenantDiscovery(boolean enabled) {
```

Comment:
If this is enabled for `TableService*Client`, would this be applicable for `TableClient` created from `getTableClient(String tableName)` method too? If yes, we should document this in the javadoc.